tardis/visualization/tools/rpacket_plot.py

```diff
@@ -0,0 +1,686 @@
+        sim : tardis.simulation.Simulation
+            simulation object generated using the run_tardis function.
+        no_of_packets : int
+            number of packets to be used for plotting.
+        """
+        self.no_of_packets = no_of_packets
+        self.sim = sim
+        self.interaction_from_num = [
+            {"text": "No Interaction", "color": "darkslategrey", "opacity": 0},
+            {"text": "e-Scattering", "color": "#3366FF", "opacity": 1},
+            {"text": "Line Interaction", "color": "#FF3300", "opacity": 1},
+        ]
+        self.theme_colors = dict(
+            light=dict(
+                linecolor="#555",
+                gridcolor="#fafafa",
+                zerolinecolor="#fafafa",
+                color="#000",
+                photosphere_line_color="black",
+                photosphere_fillcolor="darkgrey",
+                shells_line_color="black",
+                packet_line_color="darkslategrey",
+                plot_bgcolor="#fafafa",
+                paper_bgcolor="#fafafa",
+                title_font_color="#444",
+                legendgrouptitle_color="#444",
+                button_bgcolor="#fafafa",
+                button_font_color="#2A3F5F",
+                slider_font_color="#2A3F5F",
+                bordercolor="#BEC8D9",
+                slider_bgcolor="#F8FAFC",
+                slider_activebgcolor="#DBDDE0",
+                slider_currentvalue_color="#2A3F5F",
+                font_color="#000",
+            ),
+            dark=dict(
+                linecolor="#050505",
+                gridcolor="#111",
+                zerolinecolor="#111",
+                color="#fafafa",
+                photosphere_line_color="#222",
+                photosphere_fillcolor="#222",
+                shells_line_color="#555",
+                packet_line_color="#888",
+                plot_bgcolor="#000",
+                paper_bgcolor="#000",
+                title_font_color="#ccc",
+                legendgrouptitle_color="#ccc",
+                button_bgcolor="#282828",
+                button_font_color="#888",
+                slider_font_color="#888",
+                bordercolor="black",
+                slider_bgcolor="#888",
+                slider_activebgcolor="#fafafa",
+                slider_currentvalue_color="#fff",
+                font_color="#fafafa",
+            ),
+        )
+
+    @classmethod
+    def from_simulation(cls, sim, no_of_packets=15):
+        """
+        Creates an instance of RPacketPlotter from a TARDIS simulation object.
+
+        Parameters
+        ----------
+        sim : tardis.simulation.Simulation
+            TARDIS Simulation object generated using run_tardis function.
+        no_of_packets : int
+            number of packets to be used for plotting.
+
+        Returns
+        -------
+        RPacketPlotter
+        """
+        logger = logging.getLogger(__name__)
+        if hasattr(sim.runner, "rpacket_tracker_df"):
```

Comment:
As Andrew pointed out, the code has undergone significant restructuring, and .runner no longer exists. The relevant bits probably live in .transport, but you should check and update where appropriate. Shouldn't be too difficult to figure out where the new objects are located, but if you need help with specific ones, let us know in the comments.

Reply:
I am changed the runner with simulation_state and some other minor changes wherever required